general/development/tools/behat.md

```diff
@@ -0,0 +1,721 @@
+
+If you use Docker, then you may be able to use `host.docker.internal` where your site is hosted on the docker host
+
+#### Configure Behat for Moodle
+
+After setting your configuration, you can simply initialise behat:
+
+```
+$ php admin/tool/behat/cli/init.php
+```
+
+This will install all required Composer dependencies, install a new Moodle site, and put configuration in place
+
+When it finishes it will give advice on how to run Behat.
+
+#### Run Behat tests
+
+Before running behat, ensure that your Selenium server is running.
+
+The easiest way to run behat, and test that everything works is by simply running it using the command provided when you
+initialised Behat. If you didn't make a note of it, you can just run the initialisation again.
+
+```
+ $ php admin/tool/behat/cli/init.php
+```
+
+This will give you a command which you can then run. This command will run every behat scenario, which will take a considerable amount of time. This command will look a bit like this:
+
+```
+$ vendor/bin/behat --config /Users/nicols/Sites/moodles/sm/moodledata_behat/behatrun/behat/behat.yml
+```
+
+To make this more useful you an combine it with flags, for example to only run certain `tags` or for a specific Behat Feature file, or Scenario.
+
+To run all features/scenarios which are tagged with `mod_forum`:
+
+```
+    $ vendor/bin/behat --config /Users/nicols/Sites/moodles/sm/moodledata_behat/behatrun/behat/behat.yml --tags=@mod_forum
+```
+
+To run one specific feature file:
+
+```
+    $ vendor/bin/behat --config /Users/nicols/Sites/moodles/sm/moodledata_behat/behatrun/behat/behat.yml `pwd`/mod/forum/tests/behat/private_replies.feature
+```
+
+To run one specific scenario within a feature file:
+
+```
+    # To run the Scenario on line 38 of the file:
+    $ vendor/bin/behat --config /Users/nicols/Sites/moodles/sm/moodledata_behat/behatrun/behat/behat.yml `pwd`/mod/forum/tests/behat/private_replies.feature:38
+```
+
+To run one specific scenario by name:
+
+```
+    $ vendor/bin/behat --config /Users/nicols/Sites/moodles/sm/moodledata_behat/behatrun/behat/behat.yml --name="As a teacher I can see my own response"
+```
+
+See the upstream documentation on Behat, and Gherkin filters for more information.
+
+##### Running using a different browser
+
+The default browser in Behat is **Firefox**. To specify a different browser profile, you can add the `--profile` argument. For example, to use Chrome in Headless mode:
+
+```
+    $ vendor/bin/behat --config /Users/nicols/Sites/moodles/sm/moodledata_behat/behatrun/behat/behat.yml --profile=headlesschrome
+```
+
+If you are using the `moodle-browser-config` utility, then you can use any profile listed in [moodle-browser-config](/general/development/tools/behat#pre-configured-browser-profiles-moodle-browser-config). Otherwise you can write your own browser profile configuration.
```

Comment:
```suggestion
:::tip

If you are using the `moodle-browser-config` utility, then you can use any profile listed in [moodle-browser-config](/general/development/tools/behat#pre-configured-browser-profiles-moodle-browser-config). Otherwise you can write your own browser profile configuration.

:::
```